### ISSUE TITLE
networkx 2.0 breaks spectra

### ISSUE THREAD

**Issue by gfairchild** [jsvine/spectra#8]
I just updated networkx to 2.0, and this broke spectra. See the following demonstration:

```
gfairchild@meow ~/Desktop> pyvenv env
WARNING: the pyenv script is deprecated in favour of `python3.6 -m venv`
gfairchild@meow ~/Desktop> source env/bin/activate
(env) gfairchild@meow ~/Desktop> python --version
Python 3.6.3
(env) gfairchild@meow ~/Desktop> pip install spectra
Collecting spectra
  Using cached spectra-0.0.8.tar.gz
Collecting colormath (from spectra)
  Using cached colormath-2.1.1.tar.gz
Collecting numpy (from colormath->spectra)
  Using cached numpy-1.13.3-cp36-cp36m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl
Collecting networkx (from colormath->spectra)
  Using cached networkx-2.0.zip
Collecting decorator>=4.1.0 (from networkx->colormath->spectra)
  Using cached decorator-4.1.2-py2.py3-none-any.whl
Installing collected packages: numpy, decorator, networkx, colormath, spectra
  Running setup.py install for networkx ... done
  Running setup.py install for colormath ... done
  Running setup.py install for spectra ... done
Successfully installed colormath-2.1.1 decorator-4.1.2 networkx-2.0 numpy-1.13.3 spectra-0.0.8
(env) gfairchild@meow ~/Desktop> python
Python 3.6.3 (default, Oct  4 2017, 06:09:15)
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.37)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import spectra
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/gfairchild/Desktop/env/lib/python3.6/site-packages/spectra/__init__.py", line 1, in <module>
    from .core import COLOR_SPACES, Color, Scale
  File "/Users/gfairchild/Desktop/env/lib/python3.6/site-packages/spectra/core.py", line 1, in <module>
    from colormath import color_objects, color_conversions
  File "/Users/gfairchild/Desktop/env/lib/python3.6/site-packages/colormath/color_conversions.py", line 167, in <module>
    def Spectral_to_XYZ(cobj, illuminant_override=None, *args, **kwargs):
  File "/Users/gfairchild/Desktop/env/lib/python3.6/site-packages/colormath/color_conversions.py", line 159, in decorator
    _conversion_manager.add_type_conversion(start_type, target_type, f)
  File "/Users/gfairchild/Desktop/env/lib/python3.6/site-packages/colormath/color_conversions.py", line 125, in add_type_conversion
    self.conversion_graph.add_edge(start_type, target_type, {'conversion_function': conversion_function})
TypeError: add_edge() takes 3 positional arguments but 4 were given
>>> exit()
(env) gfairchild@meow ~/Desktop> pip install networkx==1.11
Collecting networkx==1.11
  Using cached networkx-1.11-py2.py3-none-any.whl
Requirement already satisfied: decorator>=3.4.0 in ./env/lib/python3.6/site-packages (from networkx==1.11)
Installing collected packages: networkx
  Found existing installation: networkx 2.0
    Uninstalling networkx-2.0:
      Successfully uninstalled networkx-2.0
Successfully installed networkx-1.11
(env) gfairchild@meow ~/Desktop> python
Python 3.6.3 (default, Oct  4 2017, 06:09:15)
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.37)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import spectra
>>> exit()
```

**Comment by jsvine** [comment]
Thanks for flagging! Looks to be an incompatibility between `networkx` and the `colormath` module. Looks like `colormath` has [a fix](https://github.com/gtaylor/python-colormath/issues/73) but that it's [not quite yet on PyPi](https://github.com/gtaylor/python-colormath/issues/79). Once the latter is resolved, I'll push a small update to `spectra` to require `colormath>=2.2.0`.

**Comment by gfairchild** [comment]
That's great news! Thanks for the update.

**Comment by NeilGirdhar** [comment]
@jsvine This is happening in color_conversions.py.  The fix is simple:

Replace

```
self.conversion_graph.add_edge(start_type, target_type, {'conversion_function': conversion_function})
```
with
```
self.conversion_graph.add_edge(start_type, target_type, conversion_function=conversion_function)
```

**Comment by jsvine** [comment]
Given that the `colormath` fix still hasn't been pushed to PyPi, I've gone ahead and pushed a new release of `spectra` (`0.0.9`), which pins the `colormath` and `networkx` requirements to the most recent compatible versions.

Running `pip install -U spectra` should now resolve the problem. Let me know if it doesn't.

**Comment by jwhite007** [comment]
@jsvine Colormath 3.0 is now out and works with NetworkX 2.0+. The dependencies of Spectra, I believe, can now be changed to reflect this. See https://pypi.python.org/pypi/colormath.

**Comment by jsvine** [comment]
Thanks! The requirements [have now been updated](https://github.com/jsvine/spectra/commit/def4bcc0868b755a9c9660a2c9eab2edeee12aa5) and are available in `v0.0.10`, available via `pip`.
